cf-agent/verify_users_pam.c

```diff
@@ -243,13 +243,14 @@ static struct spwd *GetSpEntry(const char *puser)
 }
 #endif // HAVE_FGETSPENT
 
-static bool GetPasswordHash(const char *puser, const struct passwd *passwd_info, const char **result)
+bool GetPasswordHash(const char *puser, const void *passwd_info, const char **result)
```

Comment:
I guess these are forgotten changes, judging from the comments you have commented out in `FnCallGetUserInfo()`. It's anyway wrong to call code from `cf-agent` directory, inside libpromises. Can you remove the changes in `cf-agent`?